### ISSUE TITLE
Skip 'all_admin_notices' for Gutenberg pages?

### ISSUE THREAD

**Issue by jsmoriss** [WordPress/gutenberg#6315]
When loading a Gutenberg editing page, the 'all_admin_notices' action is executed.

When saving a Gutenberg post, the 'all_admin_notices' action is executed (with or without metaboxes?).

PHP plugins may create notices, and after these notices are displayed in the 'all_admin_notices' hook, those notices are usually deleted (so they're not displayed again, and again, etc.).

Since Gutenberg hides the 'all_admin_notices' HTML, a jQuery ajax call must be made to fetch notices and display them using Gutenberg functions - unfortunately, since the 'all_admin_notices' action has already been executed (either on page load or when saving a post), those notices may no longer exist and cannot be displayed.

Is it possible to either:

- Have Gutenberg unregister all hooks from 'all_admin_notices'.

- Or have somekind of constant, function, etc. available for 'all_admin_notices' action hooks so they can check if this is a Gutenberg page and skip the display + deletion of notices?

Thanks,

js.

**Comment by jsmoriss** [comment]
FYI - for now I'm using this:

```
                        if ( function_exists( 'gutenberg_can_edit_post_type' ) &&
                                gutenberg_can_edit_post_type( $post_obj->post_type ) &&
                                        ! isset( $_REQUEST['classic-editor'] ) ) {

                                if ( ! defined( 'DOING_GUTENBERG' ) ) {
                                        define( 'DOING_GUTENBERG', true );
                                }
                        }
```

**Comment by danielbachhuber** [comment]
Related #5975 #3964

**Comment by chrisvanpatten** [comment]
Might not need documentation for this if the underlying implementation problem is addressed, but we'll wait and see…

**Comment by aduth** [comment]
Related: #11604 will attempt to detect and transparently upgrade all notices output from the server, including from `all_admin_notices.`

**Comment by chrisvanpatten** [comment]
@aduth Thanks sir! We'll close this out when that gets merged; sounds like we won't need back-compat documentation here.

**Comment by aduth** [comment]
#11604 was merged, but later (effectively) reverted in #12444. As notices made compatible turned to not be reconcilable with Gutenberg design, they are no longer planned to be inherited. See also [Notices data actions](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/data/data-core-notices.md#actions) for client-side notices documentation.

>Or have somekind of constant, function, etc. available for 'all_admin_notices' action hooks so they can check if this is a Gutenberg page and skip the display + deletion of notices?

It should be expected that [`WP_Screen::is_block_editor`](https://developer.wordpress.org/reference/classes/wp_screen/is_block_editor/) can be used for this purpose.
